flytekit/remote/remote.py

```diff
@@ -441,43 +433,42 @@ def register_workflow(
         m = OrderedDict()
         workflow_spec = get_serializable_workflow(m, serialization_settings, entity)
         ident = self._resolve_identifier(ResourceType.WORKFLOW, entity.name, version, serialization_settings)
-        self.client.create_workflow(ident, workflow_spec=workflow_spec)
-        if default_launch_plan:
-            default_lp = LaunchPlan.get_default_launch_plan(FlyteContextManager.current_context(), entity)
-            self.register_launch_plan(default_lp, serialization_settings, version=version, options=options)
-            remote_logger.debug("Created default launch plan for Workflow")
 
         if all_downstream:
             self._register_entity_if_not_exists(
                 entity, serialization_settings=serialization_settings, version=version, options=options
             )
 
+        self.client.create_workflow(ident, workflow_spec=workflow_spec)
+
+        if default_launch_plan:
+            default_lp = LaunchPlan.get_default_launch_plan(FlyteContextManager.current_context(), entity)
+            self.register_launch_plan(default_lp, version=ident.version, options=options)
+            remote_logger.debug("Created default launch plan for Workflow")
+
         return self.fetch_workflow(ident.project, ident.domain, ident.name, ident.version)
 
     def register_launch_plan(
         self,
         entity: LaunchPlan,
-        serialization_settings: typing.Optional[SerializationSettings] = None,
-        version: typing.Optional[str] = None,
+        version: str,
+        project: typing.Optional[str] = None,
```

Comment:
maybe have the serialization_settings so that we can register the workflow and tasks

Reply:
can i ignore this and leave as is since this will touch your pr anyways?